src/main/java/jimmy/mcgymmy/logic/commands/FindCommand.java

```diff
@@ -17,22 +22,78 @@
     public static final String COMMAND_WORD = "find";
     public static final String SHORT_DESCRIPTION = "Filter the displayed list by a given keyword.";
 
-    private Parameter<FoodContainsKeywordsPredicate> predicateParameter = this.addParameter(
+    private OptionalParameter<FoodContainsKeywordsPredicate> foodPredicateParameter = this.addOptionalParameter(
             "keyword",
             "",
             "keywords (case-insensitive).",
             "chicken beef mutton", (s) -> new FoodContainsKeywordsPredicate(Arrays.asList(s.split("\\s+")))
     );
+    private OptionalParameter<NameContainsKeywordsPredicate> namePredicateParameter = this.addOptionalParameter(
+            "name",
+            "n",
+            "Name of the Food",
+            "Cereal", (s) -> new NameContainsKeywordsPredicate(Arrays.asList(s.split("\\s+")))
+    );
+    private OptionalParameter<TagContainsKeywordsPredicate> tagPredicateParameter = this.addOptionalParameter(
+            "tag",
+            "t",
+            "Tag associated with the Food",
+            "Lunch", (s) -> new TagContainsKeywordsPredicate(Arrays.asList(s.split("\\s+")))
+    );
+    private OptionalParameter<DatePredicate> datePredicateParameter = this.addOptionalParameter(
+            "date",
+            "d",
+            "Date associated with the Food",
+            "20-04-2020", (s) -> new DatePredicate(s)
+    );
 
-    void setParameters(Parameter<FoodContainsKeywordsPredicate> predicateParameter) {
-        this.predicateParameter = predicateParameter;
+    void setParameters(OptionalParameter<FoodContainsKeywordsPredicate> foodPredicateParameter,
+                       OptionalParameter<NameContainsKeywordsPredicate> namePredicateParameter,
+                       OptionalParameter<TagContainsKeywordsPredicate> tagPredicateParameter,
+                       OptionalParameter<DatePredicate> datePredicateParameter) {
+        this.foodPredicateParameter = foodPredicateParameter;
+        this.namePredicateParameter = namePredicateParameter;
+        this.tagPredicateParameter = tagPredicateParameter;
+        this.datePredicateParameter = datePredicateParameter;
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        FoodContainsKeywordsPredicate predicate = predicateParameter.consume();
-        model.updateFilteredFoodList(predicate);
+        FoodContainsKeywordsPredicate foodPredicate = foodPredicateParameter.getValue().orElse(null);
+        NameContainsKeywordsPredicate namePredicate = namePredicateParameter.getValue().orElse(null);
+        TagContainsKeywordsPredicate tagPredicate = tagPredicateParameter.getValue().orElse(null);
+        DatePredicate datePredicate = datePredicateParameter.getValue().orElse(null);
+
+        Predicate<Food> combinedPredicate = null;
```

Comment:
I think for this long chain we can replace it with a stream/list of all the above predicates + a `stream.reduce` call.

Will be much cleaner

Reply:
Yeah this was initial code just to make it work, as mentioned earlier.